src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```diff
@@ -428,4 +403,115 @@ public static long getMaxDataAge(Collection<SSTableReader> sstables)
+                mergeSummary.append(i).append(':').append(mergedRowCount).append(", ");
+            }
+        }
+        mergeSummary.append('}');
+
+        StringBuilder newSSTableNames = new StringBuilder(newSStables.size() * 100);
+        for (SSTableReader reader : newSStables)
+            newSSTableNames.append(reader.descriptor.baseFilename()).append(",");
+        logger.debug(
+        "Compacted ({}) {} sstables to [{}] to level={}." +
+        " {} to {} (~{}% of original) in {}ms." +
+        " Read Throughput = {}, Write Throughput = {}, Row Throughput = ~{}/s." +
+        " {} total partitions merged to {}." +
+        " Partition merge counts were {}",
+        taskId,
+        transaction.originals().size(),
+        newSSTableNames.toString(),
+        getLevel(),
+        prettyPrintMemory(inputSizeBytes),
+        prettyPrintMemory(outputSizeBytes),
+        (int) (ratio * 100),
+        dTime,
+        prettyPrintMemoryPerSecond(inputSizeBytes, durationInNano),
+        prettyPrintMemoryPerSecond(outputSizeBytes, durationInNano),
+        totalSourceCQLRows / (TimeUnit.NANOSECONDS.toSeconds(durationInNano) + 1),
+        totalSourceRows,
+        totalKeysWritten,
+        mergeSummary.toString());
+    }
+
+    private void debugLogCompactingMessage(UUID taskId)
+    {
+        Set<SSTableReader> originals = transaction.originals();
+        StringBuilder ssTableLoggerMsg = new StringBuilder(originals.size() * 100);
+        ssTableLoggerMsg.append("Compacting (").append(taskId).append(')').append(" [");
+        for (SSTableReader sstr : originals)
+        {
+            ssTableLoggerMsg
+            .append(sstr.getFilename())
+            .append(":level=")
+            .append(sstr.getSSTableLevel())
+            .append(", ");
+        }
+        ssTableLoggerMsg.append("]");
+
+        logger.debug(ssTableLoggerMsg.toString());
+    }
+
+
+    private static void updateCompactionHistory(
+        UUID id,
+        String keyspaceName,
+        String columnFamilyName,
+        long[] mergedRowsHistogram,
+        long startSize,
+        long endSize)
+    {
+        Map<Integer, Long> mergedRows = new HashMap<>(mergedRowsHistogram.length);
+        for (int i = 0; i < mergedRowsHistogram.length; i++)
+        {
+            long count = mergedRowsHistogram[i];
+            if (count == 0)
+                continue;
+
+            int rows = i + 1;
+            mergedRows.put(rows, count);
+        }
+        SystemKeyspace.updateCompactionHistory(
+            id,
+            keyspaceName,
+            columnFamilyName,
+            System.currentTimeMillis(),
+            startSize,
+            endSize,
+            mergedRows);
+    }
+
+    private void traceLogCompactionSummaryInfo(long totalKeysWritten, long estimatedKeys, long endsize)
+    {
+        logger.trace("CF Total Bytes Compacted: {}", prettyPrintMemory(addToTotalBytesCompacted(endsize)));
```

Comment:
Checking if trace is enabled can be done here (JVM will optimize the call away) to make the caller's code a little more readable.

Reply:
I think there are 2 options - either to introduce `maybe[Trace|Debug]LogX` methods that encapsulate `if(enabled)` logic or to do the check explicitly before calling `[trace|debug]LogX` methods. 

The latter seems more consistent with the rest of the codebase, so I'd prefer to stick to it. WDYT?